packages/core/core-flows/src/inventory/workflows/create-inventory-items.ts

```diff
@@ -3,23 +3,86 @@ import {
   createWorkflow,
   transform,
 } from "@medusajs/workflows-sdk"
-import {
-  attachInventoryItemToVariants,
-  createInventoryItemsStep,
-  validateInventoryItemsForCreate,
-} from "../steps"
+import { createInventoryItemsStep } from "../steps"
 
 import { InventoryNext } from "@medusajs/types"
+import { createInventoryLevelsWorkflow } from "./create-inventory-levels"
 
+type LocationLevelWithoutInventory = Omit<
+  InventoryNext.CreateInventoryLevelInput,
+  "inventory_item_id"
+>
 interface WorkflowInput {
-  items: InventoryNext.CreateInventoryItemInput[]
+  items: (InventoryNext.CreateInventoryItemInput & {
+    location_levels?: LocationLevelWithoutInventory[]
+  })[]
+}
+
+const buildLocationLevelMapAndItemData = (data: WorkflowInput) => {
+  const inventoryItems: InventoryNext.CreateInventoryItemInput[] = []
+  // Keep an index to location levels mapping to inject the created inventory item
+  // id into the location levels workflow input
+  const locationLevelMap: Record<number, LocationLevelWithoutInventory[]> = {}
```

Comment:
I feel this index-based mapping will come to bite us in the butt at one point where the ordering of the operations isn't preserved 😄. I know we don't have much choice though, just thinking out loud.

Reply:
I'm on the same page as you are 😄 I really hate this, but there isn't an alternative here without the module accepting and returning a bulk identifier. Even in that case, you're still accepting the fact that the database is returning the right order its creating, which isn't guaranteed.

Reply:
Should we think about having a `reference_id` column where we control what is inserted and can 100% guarantee to map it back to the correspondent row?
maybe it is better to implement that now, before this happens in production

Reply:
or maybe `map_id`, something internal that we can always use in cases we need that mapping and there isn't any key yet

Reply:
We will definitely need to consider something like that @carlos-r-l-rodrigues . Let me at least open a ticket, and we can revisit this after the public preview release.

Reply:
Yes, thats the only way we can guarantee it

Reply:
but we generate the id of the entity in mikro-orm before inserting them. so there is a great chance that this is already being mapped correctly after inserting.